codeworld-server/src/Main.hs

```diff
@@ -152,7 +152,8 @@ site ctx =
             [ ("loadProject", loadProjectHandler ctx)
             , ("saveProject", saveProjectHandler ctx)
             , ("deleteProject", deleteProjectHandler ctx)
-            , ("listFolder", listFolderHandler ctx)
+            , ("directoryTree", directoryTreeHandler ctx)
+            , ("dumpTree", dumpTreeHandler ctx)
```

Comment:
If I understand correctly, the point of `dumpTree` is to store an ordering of the files and folders in a directory, so that they can be reordered?  This seems like a very heavyweight way to accomplish that goal.  Instead of serializing the entire directory structure to disk in a single file, why not store just the ordering of items in the current folder, inside `dir.info` for that directory?

The main advantage here is that it's easy to combine this with the existing directory tree.  The ordering is used only to sort lists, and everything else can be deduced from the directory structure itself.  If the directory structure doesn't match, it's okay.  You know to trust the directory structure, and just stick any items with no existing sort order at the end, or something like that.

Reply:
Ok, I'll do this.

Reply:
If `dir.info` content from

```
folder_name
```

will turn into something like

```
{
  name:  "folder_name",
  childrenOrder: ["project1", "project2", "project3", "folder1", "folder2"]
}
``` 
It looks inconvenient to use, because I should sort directories/projects not by some field, but sort their in same order as names in list.  

Also projects and folders can have intersecting names, so if one user placed project "123" before or after folder "123" - we should save this position. So ordered list like `["project", "123", "123"]` can not be source of truth.

I going to add integer order field into both `dir.info` and project json, and call `saveProject` handler on user move project.